otherlibs/unix/unix.mli

```diff
@@ -675,11 +675,39 @@ val close_process_full :
 (** {6 Symbolic links} *)
 
 
-val symlink : string -> string -> unit
-(** [symlink source dest] creates the file [dest] as a symbolic link
-   to the file [source].
-
-   On Windows: not implemented. *)
+val symlink : ?is_dir:bool -> string -> string -> unit
+(** [symlink ?is_dir source dest] creates the file [dest] as a symbolic link
+   to the file [source]. On Windows, [~is_dir] indicates if the symbolic link
+   points to a directory or a file; if omitted, [symlink] examines [source]
+   using [stat] and picks appropriately, if [source] does not exist then [false]
+   is assumed. On Unix, [~is_dir] ignored.
```

Comment:
> If it were a Windows-only implementation, I'd go for is_dir being mandatory

I'd rather say the issue is not breaking the signature of `Unix.link` too much. If the signature was designed today with a cross-platform mind I'd add the mandatory argument aswell and simply ignore it on unix. The reason is that I have the impression that in most cases (except when you implement `ln(1)`) you actually know in your program if you are linking a file or a directory and otherwise can and should `stat` for behaving correctly in a cross-platform fashion. 

So in the end maybe the initial behaviour you proposed is the best one. Old programs will compile and try to do their best on windows. New programs should use the `is_dir` argument; it could be specified in the docs that it is preferable to use the argument.

> I'm entirely easy on the name of the parameter - you came up with is_dir

Aha, bad Daniel. Yes, it usually takes me four or five iterations to find the right names... But `Unix.symlink ~dir_link` feels a little bit redundant. Why not `Unix.symlink ~to_dir` ?